war/src/main/java/io/lumeer/engine/controller/SecurityFacade.java

```diff
@@ -51,8 +52,285 @@
    private DataStorage dataStorage;
 
    @Inject
+   @SystemDataStorage
+   private DataStorage systemDataStorage;
+
+   @Inject
    private DataStorageDialect dataStorageDialect;
 
+   /********* CHECKING ROLES *********/
+
+   public boolean hasOrganizationRole(String organizationId, String user, String role) {
+      DataDocument doc = systemDataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ORGANIZATION_ROLES_COLLECTION_NAME,
+            dataStorageDialect.fieldValueFilter(LumeerConst.Security.ORGANIZATION_ID_KEY, organizationId),
+            Arrays.asList(dataStorageDialect.concatFields(LumeerConst.Security.ROLES_KEY, role)));
+
+      return checkRole(doc, user, role);
+   }
+
+   public boolean hasProjectRole(String projectId, String user, String role) {
+      DataDocument doc = dataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ROLES_COLLECTION_NAME,
+            dataStorageDialect.multipleFieldsValueFilter(projectFilter(projectId)),
+            Arrays.asList(dataStorageDialect.concatFields(LumeerConst.Security.ROLES_KEY, role)));
+
+      return checkRole(doc, user, role);
+   }
+
+   public boolean hasCollectionRole(String collectionName, String projectId, String user, String role) {
+      DataDocument doc = dataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ROLES_COLLECTION_NAME,
+            dataStorageDialect.multipleFieldsValueFilter(collectionFilter(collectionName, projectId)),
+            Arrays.asList(dataStorageDialect.concatFields(LumeerConst.Security.ROLES_KEY, role)));
+
+      return checkRole(doc, user, role);
+   }
+
+   public boolean hasViewRole(String viewId, String projectId, String user, String role) {
+      DataDocument doc = dataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ROLES_COLLECTION_NAME,
+            dataStorageDialect.multipleFieldsValueFilter(viewFilter(viewId, projectId)),
+            Arrays.asList(dataStorageDialect.concatFields(LumeerConst.Security.ROLES_KEY, role)));
+
+      return checkRole(doc, user, role);
+   }
+
+   private boolean checkRole(DataDocument rolesDocument, String user, String role) {
```

Comment:
We should not pass `user` as a parameter here but get it from `UserFacade.getUserEmail()` instead.